### PR TITLE
Demo of a chat bot using RAG.

### PR DESCRIPTION
**Pull request by cescoffier**

### CODE REVIEW

File: samples/chatbot/src/main/java/io/quarkiverse/langchain4j/sample/chatbot/ChatBotWebSocket.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Inject
+    Bot bot;
+
+    @Inject
+    ChatMemoryBean chatMemoryBean;
+
+    @OnOpen
+    public void onOpen(Session session) {
+        Infrastructure.getDefaultExecutor().execute(() -> {
```
Comment (geoand) [inline samples/chatbot/src/main/java/io/quarkiverse/langchain4j/sample/chatbot/ChatBotWebSocket.java:22]: Sneaky :)

File: samples/chatbot/src/main/java/io/quarkiverse/langchain4j/sample/chatbot/ChatMemoryBean.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.concurrent.ConcurrentHashMap;
+
+import jakarta.enterprise.context.ApplicationScoped;
+
+import dev.langchain4j.memory.ChatMemory;
+import dev.langchain4j.memory.chat.ChatMemoryProvider;
+import dev.langchain4j.memory.chat.MessageWindowChatMemory;
+
+@ApplicationScoped
+public class ChatMemoryBean implements ChatMemoryProvider {
```
Comment (geoand) [inline samples/chatbot/src/main/java/io/quarkiverse/langchain4j/sample/chatbot/ChatMemoryBean.java:13]: Does this memory work correctly with multiple users?

File: samples/chatbot/src/main/java/io/quarkiverse/langchain4j/sample/chatbot/ChatBotWebSocket.java (hunk 1 above)
Comment (cescoffier) [reply samples/chatbot/src/main/java/io/quarkiverse/langchain4j/sample/chatbot/ChatBotWebSocket.java:22]: Bad :-(

File: samples/chatbot/src/main/java/io/quarkiverse/langchain4j/sample/chatbot/ChatMemoryBean.java (hunk 2 above)
Comment (cescoffier) [reply samples/chatbot/src/main/java/io/quarkiverse/langchain4j/sample/chatbot/ChatMemoryBean.java:13]: look at the session - the Websocket Session object is your memory id.

Comment (cescoffier) [reply samples/chatbot/src/main/java/io/quarkiverse/langchain4j/sample/chatbot/ChatMemoryBean.java:13]: (Sneaky and bad)

Comment (geoand) [reply samples/chatbot/src/main/java/io/quarkiverse/langchain4j/sample/chatbot/ChatMemoryBean.java:13]: Ah, very sneaky indeed :)

**Review by geoand — APPROVED** [review]

**Pull request merged**